### PR TITLE
Fixed FastAPIServer uvicorn path

### PR DESCRIPTION
**Pull request by LeOndaz**

### CODE REVIEW

**Comment by LeOndaz** [comment]
@mrichar1 the paths were on the following format 
```
uvicorn __path__/app.py:app --host ... --port ...
```
and I removed the `.py` here.

**Comment by mrichar1** [comment]
Thanks for fixing this. For completeness we probably need tests and a sample_app for this to ensure everything runs as expected in future if there are any changes to the guvicorn/fastapi interfaces.

The steps for this are:

1. Add a fastapi sample app in `sample_apps` (based on the others).
2. Update ` test_examples/unittest_example/tests.py` to test FastAPI (based on the others).
3. Update `test_examples/pytest_example/tests.py` in the same way.

You should then be able to run `tox` and have everything pass.

**Comment by LeOndaz** [comment]
@mrichar1 Hello, I've came back from my deep coma, I'm testing in the repo because tests don't work locally and I'm lazy to debug

**Comment by LeOndaz** [comment]
@mrichar1 Have a look when you have time, this is the command that ran
```
uvicorn test_examples/unittest_example/../../sample_apps/fastapi/main:app --host 127.0.0.1 --port 8001 --reload
```
and for some reason it's not working, I'll wait for you and thanks anyway.

**Comment by mrichar1** [comment]
I've just had a quick look at this - the tests are failing as the startup methods for `uvicorn` needs the app path and app name split out.

On the command-line this would be:

`uvicorn --app-dir /home/mrichar1/git/liveandletdie/test_examples/unittest_example/../../sample_apps/fastapi/ main:app`

This will need fixed here:

https://github.com/LeOndaz/liveandletdie/blob/master/liveandletdie/__init__.py#L571-L577

To be something like:

```
def create_command(self):
        path = os.path.dirname(self.path)

        return [
            self.executable,
            '--app-dir',
            path,
            'main:app',
            '--host',
            self.host,
            '--port',
            self.port,
            '--reload',
        ]
```

Additionally, from looking at the `uvicorn` docs, it looks like `request.url` will be an object, so here:

https://github.com/LeOndaz/liveandletdie/blob/master/sample_apps/fastapi/main.py#L10

You need to instead do:

```
if request.url.scheme == 'https':
```

that should get the tests passing, though you'll probably want to also test that the instance of `uvicorn` that `liveandletdie` is starting is correctly configured for real-world usage.

File: sample_apps/fastapi/main.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# encoding: utf-8
+from fastapi import FastAPI, Request
+
+app = FastAPI()
+
+
+@app.get('/')
+def home(request: Request):
+    content = 'Home FastAPI'
+    if request.url.startswith('https://'):
```
Comment (jensens) [inline sample_apps/fastapi/main.py:10]:
```suggestion
    if request.url.scheme == 'https':
```

**Comment by jensens** [comment]
followup at #16

**Review by LeOndaz — COMMENTED** [review]
LGTM

**Pull request closed without merging**